### ISSUE TITLE
GitHub work flow and continuous integration of R packages

### ISSUE THREAD

**Issue by maurolepore** [forestgeo/learn#1]
To develop our software I recommend to follow the [GitHub flow](https://help.github.com/articles/github-flow/):

1. Create a branch from the repository.
2. Create, edit, rename, move, or delete files.
3. Send a pull request from your branch with your proposed changes to kick off a discussion.
4. Make changes on your branch as needed. Your pull request will update automatically.
5. Merge the pull request once the branch is ready to be merged.
6. Tidy up your branches using the delete button in the pull request or on the branches page.

This is, I recommend to collaborate using the **_shared repository_** [development model](https://help.github.com/articles/about-collaborative-development-models/) (simplest):

> In the shared repository model, collaborators are granted push access to a single shared repository and topic branches are created when changes need to be made. Pull requests are useful in this model as they initiate code review and general discussion about a set of changes before the changes are merged into the main development branch. This model is more prevalent with small teams and organizations collaborating on private projects.

This approach allows [continuous integration](http://searchsoftwarequality.techtarget.com/definition/continuous-integration)

> Continuous integration (CI) is a software engineering practice in which isolated changes are immediately tested and reported on when they are added to a larger code base. The goal of CI is to provide rapid feedback so that if a defect is introduced into the code base, it can be identified and corrected as soon as possible. Continuous integration software tools can be used to automate the testing and build a document trail.

Continuous integration via [TravisCI](https://docs.travis-ci.com/user/for-beginners) is highly recommended to [automatically check R Packages after every commit](http://r-pkgs.had.co.nz/check.html#travis).

> If you use git and GitHub ... I highly recommend learning about Travis. Travis is a continuous integration service, which means that it runs automated testing code everytime you push to GitHub. ... For an R package, the most useful code to run is devtools::check().
